metaflow/plugins/kfp/kfp.py

```diff
@@ -231,15 +231,23 @@ def create_workflow_yaml(self, output_path: str, output_format: str = "kfp") ->
         if output_format == "kfp":
             pass  # KFP pipeline yaml is created by default
         elif output_format == "argo-workflow":
-            workflow["spec"]["serviceAccountName"] = (
-                KUBERNETES_SERVICE_ACCOUNT or "default-editor"
-            )
+            # Output of KFP compiler already has workflow["kind"] = "Workflow".
+
+            # Keep generateName - Argo Workflow is usually used in single run.
+
+            # Service account is added through webhooks.
+            workflow["spec"].pop("serviceAccountName", None)
+
         elif output_format == "argo-workflow-template":
-            workflow["spec"]["serviceAccountName"] = (
-                KUBERNETES_SERVICE_ACCOUNT or "default-editor"
-            )
             workflow["kind"] = "WorkflowTemplate"
-            workflow["status"] = None
+
+            # Use static name to make referencing easier.
+            workflow["metadata"]["name"] = (
+                workflow["metadata"].pop("generateName").rstrip("-")
```

Comment:
It'd be beneficial to document how you are using the kfp sanitized k8s workflow name

Reply:
Added as doc string.